participant/v1/participant.proto

```diff
@@ -110,70 +110,114 @@ message GetNotaryRequest {
 // Responds with the signature of the requested Notary
 /* [Example]
 {
-  "signature": "ByteLengthSignatureVariesByAlgo="
+  "signature": "ByteLengthSignatureVariesByAlgo"
 }
 */
 message GetNotaryResponse {
   // The signature of the Notary created by signing the Notary Public key with the Authority keypair
   bytes signature = 1;
 }
 
-// Requests the count of files of various denominations for a given currency.
-/* [Example]
+// Request to sign a contract
+/* [Example}
 {
-  "asset_code": "USD"
+  "unsigned_contract": "",
 }
 */
-message GetSupplyRequest {
-  // The currency code of the requested supply.
-  string asset_code = 1;
-  // [MultibaseVerifier] Treasury verifier.
-  string verifier   = 2;
+message SignContractRequest {
+  // Unsigned contract ttl
+  string unsigned_contract = 1;
 }
 
-// Lists the count of the files of specific denominations held in the Treasury service's vault, and the total value held
-// in files of that denomination.
-/* [Example]
+// Response of a signed contract
+/* [Example}
 {
-  "denomination": {
-    "asset_code":"USD",
-    "amount":2000,
-    "decimals":2
-  },
-  "count": 5,
-  "total": {
-    "asset_code":"USD",
-    "amount":10000,
-    "decimals":2
-  }
+  "signed_contract": "",
 }
 */
-message DenominationCount {
-  // The amount of the denomination.
-  common.Amount denomination = 1;
-  // The number of digital banknotes of this amount held in supply.
-  uint32 count               = 2;
-  // The total value represented by digital banknotes of this denomination.
-  common.Amount total        = 3;
+message SignContractResponse {
+  // Signed contract ttl
+  string signed_contract = 1;
 }
 
-// Provides a list of file denominations in the Treasury service's Vault.
-/* [Example]
+// Request to create an asset
+/* [Example}
 {
-  "total": {
+  "authority_verifier": "AuthorityVerifierLengthVariesByAlgo",
+  "recipient_verifier": "RecipientVerifierLengthVariesByAlgo",
+  "group": "g1",
+  "amount": {
     "asset_code": "USD",
-    "amount": 15000,
-    "decimals": 2
-  }
+    "value": "123.000000"
+  },
+  "created_at": "1970-01-01T00:00:00.00Z"
+}
+*/
+message CreateAssetRequest {
+  // Authority that creates the asset
+  string authority_verifier           = 1;
+  // Recipient of the created asset
+  string recipient_verifier           = 2;
+  // Group
+  string group                        = 3;
+  // Amount
+  common.Amount amount                = 4;
+  // Timestamp
+  string timestamp                    = 5;
+}
+
+// Response to create an asset
+/* [Example}
+{
+  "transferred_asset": "ByteLengthTransferredAssetVariesByAlgo"
 }
 */
-message GetSupplyResponse {
-  // A list of denomination counts of the digital banknotes held in supply.
-  reserved 1;
-  reserved "supply";
+message CreateAssetResponse {
+  // An asset that has been transferred to a new owner, but not yet authorized
```

Comment:
Should this comment be referencing asset creation instead of transfers, or is this also accurate for creation?